src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/cache/DefaultCache.java

```diff
@@ -145,22 +157,93 @@ public boolean matchCacheableRequest(
         final DefaultCacheEntry cacheEntry = cachedEntries.get(requestHash);
         return satisfiedByCache(requestHeaders) &&
                 cacheEntry != null &&
-                cacheEntry.canServeRequest(requestHeaders, authScope);
+                cacheEntry.canServeRequest(requestHeaders, authScope) &&
+                !cacheEntry.isValidationRequired();
 
     }
 
 
     public void purge(
         int requestHash)
     {
-        DefaultCacheEntry cacheEntry = this.cachedEntries.remove(requestHash);
-        entryCount.accept(-1);
+        DefaultCacheEntry cacheEntry = cachedEntries.remove(requestHash);
         if (cacheEntry != null)
         {
+            entryCount.accept(-1);
             cacheEntry.purge();
         }
+        links.remove(requestHash);
+    }
+
+    public void invalidateCacheEntryIfNecessary(
+        int requestHash,
+        String requestUrl,
+        ArrayFW<HttpHeaderFW> headers)
+    {
+        DefaultCacheEntry cacheEntry = cachedEntries.remove(requestHash);
+        if (cacheEntry != null)
+        {
+            cacheEntry.setValidationRequired(true);
+        }
+
+        headers.forEach(header ->
+        {
+            invalidateLinkCacheEntry(requestUrl, header);
+        });
     }
 
+    private void invalidateLinkCacheEntry(
+        String requestUrl,
+        HttpHeaderFW header)
+    {
+        if (LINK.equals(header.name().asString()))
+        {
+            Matcher matcher = LINK_URL_PATTERN.matcher(header.value().asString());
+            if (matcher.find())
+            {
+                String linkFullURL = matcher.group(1);
+                if (linkFullURL != null)
+                {
+                    try
+                    {
+                        final URI requestURI = new URI(requestUrl);
+                        if (URL_PATTERN.asPredicate().test(linkFullURL))
+                        {
+                            final URI linkRequestURI = new URI(linkFullURL);
+                            if (!requestURI.getAuthority().equals(linkRequestURI.getAuthority()))
+                            {
+                                return;
+                            }
+                        }
+                        else
+                        {
+                            linkFullURL = String.format("%s://%s%s",
+                                                        requestURI.getScheme(),
+                                                        requestURI.getAuthority(),
+                                                        linkFullURL);
```

Comment:
I think this is implying that `linkFullURL` is just a path, but it is a bit difficult to follow the logic.